Add KubeFlow Operator

Tomcli opened the pull request:
Add KubeFlow Operator to OperatorHub 

cc @animeshsingh

Signed-off-by: Tommy Li <Tommy.Chaoping.Li@ibm.com>

Thanks submitting your Operator. Please check below list before you create your Pull Request.
*************************************************
**Flat operator directory structure is obsolete from 23-rd of October 2019, only nested directory structure will be accepted.**
*************************************************

### New Submissions

* [x] Has you operator [nested directory structure](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#create-a-bundle)?
* [x] Have you selected the Project *Community Operator Submissions* in your PR on the right-hand menu bar?
* [x] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md)?
* [x] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [x] Have you tested your Operator with all Custom Resource Definitions?
* [x] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?

### Updates to existing Operators

* [ ] Is your new CSV pointing to the previous version with the `replaces` property?
* [ ] Is your new CSV referenced in the [appropriate channel](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#bundle-format) defined in the `package.yaml` ?
* [ ] Have you tested an update to your Operator when deployed via OLM?
* [ ] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?

### Your submission should not

* [ ] Modify more than one operator
* [ ] Modify an Operator you don't own
* [ ] Rename an operator - please remove and add with a different name instead
* [ ] Submit operators to both `upstream-community-operators` and `community-operators` at once
* [ ] Modify any files outside the above mentioned folders
* [ ] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [x] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file a PR against this repo and explain your need

<sup>2</sup> For more information see [here](https://github.com/operator-framework/operator-sdk/blob/master/doc/images/operator-capability-level.svg)

animeshsingh commented:
/lgtm

J0zi commented:
@Tomcli, please check following error:
installplan status for "null": Error from server (NotFound): installplans.operators.coreos.com "null" not found

J0zi commented on upstream-community-operators/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,107 @@
+  description: "Kubeflow Operator for deployment and management of Kubeflow components. Applicable for Kubeflow versions 0.7 and above. Check [KubeFlow Operator documentation](https://github.com/kubeflow/kfctl/blob/master/operator.md) for more details."
+  displayName: Kubeflow
+  icon: 
+  - base64data: iVBORw0KGgoAAAANSUhEUgAABvAAAAbwCAYAAAC1MBbrAAAACXBIWXMAAC4jAAAuIwF4pT92AAAgAElEQVR4nOzdPXJbV6Ku4W+fcngD3hEc3hEcMt5Bgz0CaQQg81VlKuxIUnRDUVXIhT0CcQRXcIBY8AgMzwBB5/sGG2pL3f6RBIALG3ieKpVY7bb5lX9kF1+utRIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA ...
```
'replaces' property is pointing to non-existing operator kfctl.v0.0.0, please remove the line

Tomcli commented:
Thanks @J0zi, I think the issue was the crd file naming. We fixed the file naming and added few selector to match with the operator. The CI test should be passing now.

J0zi commented:
Great job @Tomcli! Please squash your commits to one.

Tomcli commented:
Thanks @J0zi, I have squashed my commits.
